tests/unit/kms.js

```diff
@@ -0,0 +1,181 @@
+            masterKeyId;
+        kms.bucketLevelEncryption(
+            'dummyBucket', headers, log,
+            (err, sseInfo) => {
+                kms.createCipherBundle(
+                    sseInfo, log, (err, cipherBundle) => {
+                        assert.strictEqual(cipherBundle.algorithm,
+                                           sseInfo.algorithm);
+                        assert.strictEqual(cipherBundle.masterKeyId,
+                                           sseInfo.masterKeyId);
+                        assert.strictEqual(cipherBundle.cryptoScheme,
+                                           sseInfo.cryptoScheme);
+                        assert.notEqual(cipherBundle.cipheredDataKey, null);
+                        assert.notEqual(cipherBundle.cipher, null);
+                        done();
+                    });
+            });
+    });
+
+    /* cb(err, cipherBundle, decipherBundle)*/
+    function _utestCreateBundlePair(log, cb) {
+        const algorithm = 'AES256';
+        const headers = {
+            'x-amz-scal-server-side-encryption': algorithm,
+        };
+        kms.bucketLevelEncryption(
+            'dummyBucket', headers, log,
+            (err, sseInfo) => {
+                if (err) {
+                    cb(err);
+                    return;
+                }
+                kms.createCipherBundle(
+                    sseInfo, log, (err, cipherBundle) => {
+                        if (err) {
+                            cb(err);
+                            return;
+                        }
+                        sseInfo.cipheredDataKey =
+                            new Buffer(cipherBundle.cipheredDataKey, 'base64');
+                        kms.createDecipherBundle(
+                            sseInfo, 0, log, (err, decipherBundle) => {
+                                if (err) {
+                                    cb(err);
+                                    return;
+                                }
+                                assert.strictEqual(typeof decipherBundle,
+                                                   'object');
+                                assert.strictEqual(decipherBundle.cryptoScheme,
+                                                   cipherBundle.cryptoScheme);
+                                assert.notEqual(decipherBundle.decipher, null);
+                                cb(null, cipherBundle, decipherBundle);
+                            });
+                    });
+            });
+    }
+
+    it('should cipher and decipher a datastream', done => {
+        _utestCreateBundlePair(log, (err, cipherBundle, decipherBundle) => {
+            assert.strictEqual(err, null);
+            cipherBundle.cipher.pipe(decipherBundle.decipher);
+            const target = new Buffer(10000);
+            target.fill('e');
+            cipherBundle.cipher.write(target);
+            const result = decipherBundle.decipher.read();
```

Comment:
I believe this will just read what is contained in the readable stream buffer.  So, you are relying on the high water mark default size of 16 KB for the test to pass.  You should note that or put a listener on the stream and concatenate the result.